go/types/opcache.go

```diff
@@ -91,37 +125,100 @@ func encToSlice(v Value, initBuf []byte, vw ValueWriter) []byte {
 	return w.data()
 }
 
-func (p *opCache) NewIterator() *opCacheIterator {
-	return &opCacheIterator{p.ops.NewIterator(nil, nil), p.vrw}
+func (opc *ldbOpCache) NewIterator() opCacheIterator {
+	prefix := [4]byte{}
+	binary.LittleEndian.PutUint32(prefix[:], opc.colId)
+	return &ldbOpCacheIterator{iter: opc.store.ldb.NewIterator(util.BytesPrefix(prefix[:]), nil), vr: opc.vrw}
 }
 
-func (p *opCache) Destroy() error {
-	d.Chk.NoError(p.ops.Close())
-	return os.RemoveAll(p.dbDir)
-}
-
-func (i *opCacheIterator) Next() bool {
+func (i *ldbOpCacheIterator) Next() bool {
 	return i.iter.Next()
 }
 
-func (i *opCacheIterator) Op() sequenceItem {
+func (i *ldbOpCacheIterator) MapOp() sequenceItem {
 	entry := mapEntry{}
 	ldbKey := i.iter.Key()
-	data := i.iter.Value()
-	dataOffset := 0
-	switch NomsKind(ldbKey[0]) {
+	ldbValue := i.iter.Value()
+	switch NomsKind(ldbKey[uint32Size]) {
 	case BoolKind, NumberKind, StringKind:
-		entry.key = DecodeFromBytes(ldbKey, i.vr, staticTypeCache)
+		entry.key = DecodeFromBytes(ldbKey[4:], i.vr, staticTypeCache)
```

Comment:
can this `4` be a constant of some kind? So if we change the length of the prefix we use at some point, we don't need to try to find all the places we assumed it was 4?